diff --git a/src/glsl/nir/nir_types.h b/src/glsl/nir/nir_types.h
index a8ff8f2..bf2faf4 100644
--- a/src/glsl/nir/nir_types.h
+++ b/src/glsl/nir/nir_types.h
@@ -27,6 +27,8 @@
 
 #pragma once
 
+#include <stdio.h>
+
 /* C wrapper around glsl_types.h */
 
 #include "../glsl_types.h"
@@ -37,8 +39,6 @@ extern "C" {
 struct glsl_type;
 #endif
 
-#include <stdio.h>
-
 void glsl_print_type(const struct glsl_type *type, FILE *fp);
 void glsl_print_struct(const struct glsl_type *type, FILE *fp);
 
